src/Core/Billing/Services/Implementations/SubscriberService.cs

```diff
@@ -609,25 +610,54 @@ public async Task UpdateTaxInformation(
             }
         });
 
-        if (!subscriber.IsUser())
+        var taxId = customer.TaxIds?.FirstOrDefault();
+
+        if (taxId != null)
+        {
+            await stripeAdapter.TaxIdDeleteAsync(customer.Id, taxId.Id);
+        }
+
+        if (string.IsNullOrWhiteSpace(taxInformation.TaxId))
         {
-            var taxId = customer.TaxIds?.FirstOrDefault();
+            return;
```

Comment:
Let's add test coverage for the new paths.

Reply:
There is no point, we have to refactor this anyways. There's other requirements we might want to get in for the January 7 release. I can comment on it privately if you like.